python/revng/cli/_commands/ptml.py

```diff
@@ -0,0 +1,273 @@
+
+
+def cat_ptml_plain_many(content: Dict[str, str]):
+    for key, value in content.items():
+        print(f"{key}:")
+        print(re.sub("\n", "\n  ", f"  {strip_ptml(value)}"))
+
+
+def cat_ptml_color_single(content: str, console):
+    dom = parseString(content)
+    _cat_ptml_color(dom, console, "", {})
+
+
+def cat_ptml_color_many(content: Dict[str, str], console):
+    for key, value in content.items():
+        console.print(f"{key}:", style="yellow1")
+        dom = parseString(value)
+        console.print("  ", end="")
+        _cat_ptml_color(dom, console, "  ", {})
+        console.print("\n", end="")
+
+
+def _cat_ptml_color(node: Document, console, indent: str, metadata: Dict[str, str]):
+    for node in node.childNodes:
+        if node.nodeType == Node.TEXT_NODE:
+            content = re.sub("\n", f"\n{indent}", node.nodeValue)
+            if "data-token" in metadata and metadata["data-token"] in COLOR_CONVERSION:
+                console.print(content, end="", style=COLOR_CONVERSION[metadata["data-token"]])
+            else:
+                console.print(content, end="")
+        elif node.nodeType == Node.ELEMENT_NODE:
+            new_metadata = {**metadata}
+            for key, value in node.attributes.items():
+                new_metadata[key] = value
+            _cat_ptml_color(node, console, indent, new_metadata)
+        else:
+            pass
+
+
+def cmd_cat(args):
+    if args.color and not color:
+        print("Module 'rich' not found, please install it to use color mode")
+        sys.exit(1)
+
+    filters = normalize_filter_extract(args.filter, args.extract)
+    content = args.input.read()
+    if args.plain or not color:
+        handle_file(content, cat_ptml_plain_single, cat_ptml_plain_many, filters)
+    else:
+        console = Console(markup=False, highlight=False, force_terminal=args.color)
+        console.options.no_wrap = True
+        handle_file(
+            content,
+            lambda x: cat_ptml_color_single(x, console),
+            lambda x: cat_ptml_color_many(x, console),
+            filters,
+        )
+
+
+def cmd_strip(args):
+    if args.inplace and args.input == sys.stdin:
+        print("Cannot strip inplace while reading from stdin", file=sys.stderr)
+        sys.exit(1)
+
+    filters = normalize_filter_extract(args.filter, args.extract)
+    content = args.input.read()
+    result = handle_file(content, strip_ptml, strip_ptml_many, filters)
+    if args.inplace:
+        args.input.seek(0)
+        args.input.truncate(0)
+        args.input.write(result)
+    else:
+        args.output.write(result)
+
+
+class PTMLCommand(Command):
+    def __init__(self):
+        super().__init__(("ptml",), "Tool to manipulate PTML files")
+
+    def register_arguments(self, parser: argparse.ArgumentParser):
+        subparsers = parser.add_subparsers(title="Subcommands", metavar="")
+
+        parser_cat = subparsers.add_parser("cat", help="Print the ptml on the console")
```

Comment:
No wait, we already have a way to handle subcommands, pretty please use that one.

Make distinct commands:
```python
super().__init__(("ptml","cat"), "Tool to manipulate PTML files")
```

Reply:
Done